x-pack/test/functional/services/transform/discover.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function TransformDiscoverProvider({ getService }: FtrProviderContext) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+
+  return {
+    async assertDiscoverQueryHits(expectedDiscoverQueryHits: string) {
+      await testSubjects.existOrFail('discoverQueryHits');
+
+      const actualDiscoverQueryHits = await testSubjects.getVisibleText('discoverQueryHits');
+
+      expect(actualDiscoverQueryHits).to.be(expectedDiscoverQueryHits);
```

Comment:
Would be good to add an assertion message, otherwise a failure would be something like `expected 5 to kind of equal 7`, which is not super helpful when looking into failed tests. Also, we usually don't use `be` as it doesn't have support for custom assertion messages. Instead we mostly go with `eql`.
Maybe something like:
```suggestion
      expect(actualDiscoverQueryHits).to.eql(expectedDiscoverQueryHits, `Discover query hits should be ${expectedDiscoverQueryHits}, got ${actualDiscoverQueryHits}`);
```

Reply:
Fixed in 2616008.